src/Cms/Item.php

```diff
@@ -23,6 +23,9 @@
  */
 class Item
 {
+	/**
+	 * @use HasSiblings<self::ITEMS_CLASS>
```

Comment:
It somehow seemed to work in some cases. but then it also doesn't. Psalm seems to be buggy there. I added an issue with them as well. There also only is `self` but not `static` but `self` seems to act more like `static`. But you're right, we probably should just replace it with hardcoded  values.... though I haven't found a good way for a default template value:

How can I tell Psalm: use `Items` when no template is passed, but use the passed template when a child class extends this class?

Reply:
I don't think a child class can override the template type without also importing the trait a second time. Where would we put the `@use` rule in the child class otherwise?

If `self` really means `static` here, then keeping it dynamic would of course be even better. It's just quite confusing, so we might need an extra comment line explaining that.

Reply:
I think `Item` would need to add a new class template which it passes to HasSiblings. And then the classes extending `Item` need to pass the actual type via `@extends Item<Blocks>` (for example). This would work well except that for the `Item` class itself the template would be `mixed` as I don't think one can set a default template value of `Items` (as in "this template is Items as long as any child doesn't pass it's more specific type for the template").

Reply:
Don't think templates can have default values. But would have been great, I agree.